### PR TITLE
Organization CRUD operations

### PR DESCRIPTION
**Pull request by TehanovAnton**

### CODE REVIEW

**Comment by TehanovAnton** [comment]
#9 I add associations and scaffold for Organization model. Implement CRUD operations on Organization in controller

File: app/controllers/cities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  def destroy
+    @city.destroy
+    respond_to do |format|
+      format.html { redirect_to cities_url, notice: "City was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
```
Comment (dmokreckiy) [inline app/controllers/cities_controller.rb]: Removed obsolete comments

File: app/controllers/cities_controller.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+class CitiesController < ApplicationController
+  before_action :set_city, only: %i[ show edit update destroy ]
+
+  def index
+    @cities = City.all
```
Comment (dmokreckiy) [inline app/controllers/cities_controller.rb:7]: Sidenote for further development - needs pagination

File: app/controllers/organizations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        format.json { render :show, status: :ok, location: @organization }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @organization.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @organization.destroy
```
Comment (dmokreckiy) [inline app/controllers/organizations_controller.rb]: Destory can fail and return false you know - https://devdocs.io/rails~6.1/activerecord/persistence#method-i-destroy

File: app/models/organization.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 class Organization < ApplicationRecord
-  has_many :cities_organizations
+  has_many :cities_organizations, dependent: :destroy
   has_many :cities, through: :cities_organizations
+
+  validates :adress, :email, :name, :phone_number, presence: true
```
Comment (dmokreckiy) [inline app/models/organization.rb]: You should be able to combine these two into one
`validates :adress, :email, :name, :phone_number, presence: true, uniqueness: true`

File: db/migrate/20211217150125_add_omniauth_to_users.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (dmokreckiy) [inline db/migrate/20211217150125_add_omniauth_to_users.rb:1]: Migrations almost always one time runners - no need to use `# frozen_string_literal: true`

File: app/services/organization_service.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class OrganizationService
```
Comment (dmokreckiy) [inline app/services/organization_service.rb]: OK, this is wrong on so many levels...

File: app/services/organization_service.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  def save_organization_and_cities_organization(organization: Organization.new,
```
Comment (dmokreckiy) [inline app/services/organization_service.rb]: Almost any service should have `def initialize` where you prepare all instance variables you pass on to a service. If not - this is a utility library (some calculations, etc).

File: app/services/organization_service.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+                                                 city_ids: [])
+    return false unless organization.valid?
+
+    organization.save
+
+    city_ids.each do |city_id|
+      CitiesOrganization.create(city_id: city_id, organization_id: organization.id)
```
Comment (dmokreckiy) [inline app/services/organization_service.rb]: You should not have to create an intermediate records - Rails will take care of that. Take a closer look at `physician.patients = patients` https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association

File: app/services/organization_service.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    end
+  end
+end
```
Comment (dmokreckiy) [inline app/services/organization_service.rb]: You don't even need this service. Rails will do all of these automatically with properly setuped associations. Also check out how are Forms created in Rails https://guides.rubyonrails.org/form_helpers.html

**Review by dmokreckiy — CHANGES_REQUESTED** [review]
Please rewrite OrganizationService with built-in Rails methods.

**Comment by dmokreckiy** [comment]
@TehanovAnton No PR description

File: .rubocop.yml [hunk 10]
```diff
@@ -3,6 +3,9 @@ require: rubocop-rails
 Style/Documentation:
   Enabled: false
 
-Rails:
+AllCops:
   Exclude:
-    - 'config/**/*'
\ No newline at end of file
+    - config/**/*
```
Comment (dmokreckiy) [inline .rubocop.yml:8]: :+1:

**Review by yury-yaroshyk-itechart — APPROVED** [review]

**Pull request merged**